daemon/graphdriver/quota/projectquota.go

```diff
@@ -194,6 +247,49 @@ func setProjectQuota(backingFsBlockDev string, projectID uint32, quota Quota) er
 	return nil
 }
 
+// Ext4Quota is a struct implements quota operations
+type Ext4Quota struct {
+}
+
+// SetProjectQuota - set the quota for project id on ext4 block device
+func (q *Ext4Quota) SetProjectQuota(backingFsBlockDev string, projectID uint32, quota Quota) error {
```

Comment:
@amir73il  Thank you for your response and sorry for the late response :)
>　Am I wrong to assume that this is not really the "Ext4Quota" implementation, but the new generic "VfsQuota" implementation, which can also be applied to other fs that may support project quota in the future?

It's true that is `VfsQuota`, I named it to `Ext4Quota` because it's only used for ext4 at the moment, I'm fine to rename it to `VfsQuota`

>Am I wrong to assume that since kernel version v4.5, both xfs and ext4 supports that same unified API?
Am I wrong to assume that if you need to use Q_SETPQUOTA, you also need to use Q_GETPQUOTA and Q_GETPQSTAT for ext4?
If I am not wrong, I suggest to determine the QuotafileOps to use by kernel version (either XFS legacy or VFS ops) and to include also GetProjectQuota() and GetProjectQuotaStat() in the QuotafileOps struct.

The main reason I  implemented like this is we have a backported kernel (3.10 with some ext4 project quota)，the API for Ext4 and XFS are not the same unified API.
 Because of the backported kernel, determining the QuotafileOpts to use by kernel version is a hard thing. 
For `GetQuotaStat`,  from what I know, the `vfsQuota` hasn't implemented it yet(https://github.com/torvalds/linux/blob/master/fs/quota/quota.c#L718), so we still need to use `Q_XGETQSTAT`.  

So how about use `Q_GETQUOTA` and `Q_SETQUOTA` in `QuotafileOps` for ext4 and use `Q_XSETQLIM` and `Q_XGETQUOTA` for xfs?